### DISCUSSION TITLE
Disabling processing of a certain .js file

### DISCUSSION THREAD

**Discussion by utdemir** [parcel-bundler/parcel#5806] [Q&A]
I have an externally produced javascript file that I import on my `index.html`.

Currently, parcel transform that javascript file adding a boilerplate at the beginning. However, this breaks the script because it sets `var process = require("process");` and that breaks the script I have (because it checks the `process` variable to see if it's running on browser or nodejs.

So, I was wondering if there's a way to skip processing the module and import them as-is. Alternatively, have a way to not insert the boilerplate at the beginning (I don't even understand why is it doing that in case of a JS file which is imported as: `<script src="../assets/foo.js"></script>`)

**Comment by danieltroger** [comment]
Can't you detect if you're node differently? I.e. by checking if document.body exists?

**Reply by utdemir** [reply]
Unfortunately that is a horribly unreadable auto-generated javascript that I do not want to patch.

**Reply by danieltroger** [reply]
Ah, that sucks. I have not 100% understood your problem but can you check https://v2.parceljs.org/features/node-emulation/, maybe that helps?

If I understand it correctly you don't want parcel to emulate some node related process thing, and IIRC it is described the the linked document how you can turn node emulation off

**Reply by mischnic** [reply]
Automatically polyfilling `process` can't be disabled at the moment

**Comment by utdemir** [comment]
> can you check https://v2.parceljs.org/features/node-emulation/, maybe that helps?

Thank you @danieltroger , that was promising but unfortunately it didn't change the polyfill, as @mischnic mentioned later on.

I ended up "fixing" it by prepending "var process = undefined;" in front of the script :/. It's pretty ugly, but works.

Thanks for your help.
